doc/quotas.md

```diff
@@ -0,0 +1,110 @@
+# Quotas
+
+In large Graphite (go-carbon) storage installations, it's desirable to have control how many resources a user can consume.
+
+Go-Carbon addresses this issue by the implementation pattern-matching based quota design.
+
+Limitations/Caveats:
+
+* The current implementation is based on concurrent/realtime trie-index, so to use quotas, also need to enable those features on go-carbon.
+* Pattern matching rules aren't regexp, but like graphite query syntax, using globs
+* Quota and usage metrics are only produced if
+
+## Available controls
+
+* namespaces: how many sub-namespaces are allowed for the matched prefix/namespaces
```

Comment:
@bom-d-van It is not entirely clear to me what use-case does this cover. Why do we need this?

Reply:
For namespaces, it can be used to limit the number of top namespaces allowed to create in the cluster/instance.
For example, limit the number of namespaces under `sys.` to 10 give go-carbon administrator control to avoid users pollute the top metric namespaces.